src/Domain.LinnApps/PurchaseOrders/PurchaseOrderService.cs

```diff
@@ -0,0 +1,29 @@
+﻿namespace Linn.Purchasing.Domain.LinnApps.PartSuppliers
+{
+    using System.Collections.Generic;
+
+    using Linn.Common.Authorisation;
+    using Linn.Purchasing.Domain.LinnApps.PurchaseOrders;
+
+    public class PurchaseOrderService : IPurchaseOrderService
+    {
+        private readonly IAuthorisationService authService;
+
+        public PurchaseOrderService(
+            IAuthorisationService authService)
+        {
+            this.authService = authService;
+        }
+
+        public void UpdatePurchaseOrder(PurchaseOrder current, PurchaseOrder updated, IEnumerable<string> privileges)
+        {
+            if (!this.authService.HasPermissionFor(AuthorisedAction.PartSupplierUpdate, privileges))
```

Comment:
should be some sort purchasing team privilege?

Reply:
we should probably  decide on a more uniform approach to privileges since its all pretty ad hoc at the moment. Can discuss at show and tell I'm going pretty granular with privileges, e.g. part-supplier.update, part-supplier.create, purchase-order.overbook etc, and then grouping them all via a purchasing.admin group.

Reply:
http://app.linn.co.uk/authorisation/groups/9